src/rpc/rawtransaction.cpp

```diff
@@ -858,132 +965,180 @@ UniValue consolidatemsunspent(const UniValue& params, bool fHelp)
+        // Not adding the vout of 37 bytes here since it must be subtracted from both side of equation to balance.
+        nBase = (38 + ((74 * vOpCodes[0]) + 2) + ((int)nRedeemScriptSize + 4));
+        dEqResult = (15000 - 37) / (double)nBase;
 
-    while (pblkindex->nHeight < nBlockEnd)
-    {
-        if (fComplete)
-            break;
+        // Round up
+        int nEqMaxInputs = std::ceil(dEqResult);
 
-        pblkindex = pblkindex->pnext;
+        if (nMaxInputs == 0 || nMaxInputs > nEqMaxInputs)
+            nMaxInputs = nEqMaxInputs;
 
-        CBlock block;
+        GRC::BlockFinder blockfinder;
 
-        if (!block.ReadFromDisk(pblkindex, true))
-            throw JSONRPCError(RPC_PARSE_ERROR, "Unable to read block from disk!");
+        CBlockIndex* pblkindex = blockfinder.FindByHeight((nBlockStart - 1));
 
-        for (unsigned int i = 1; i < block.vtx.size(); i++)
+        if (!pblkindex)
+            throw JSONRPCError(RPC_PARSE_ERROR, "Block not found");
+
+        bool fComplete = false;
+
+        while (pblkindex->nHeight < nBlockEnd)
         {
             if (fComplete)
                 break;
 
-            // Load Transaction
-            CTransaction tx;
-            CTxDB txdb("r");
-            CTxIndex txindex;
-            uint256 hash;
+            pblkindex = pblkindex->pnext;
+            nBlockCurrent = pblkindex->nHeight;
 
-            hash = block.vtx[i].GetHash();
+            CBlock block;
 
-            // In case a fail here we can just continue thou it shouldn't happen
-            if (!ReadTxFromDisk(tx, txdb, COutPoint(hash, 0), txindex))
-                continue;
+            if (!block.ReadFromDisk(pblkindex, true))
+                throw JSONRPCError(RPC_PARSE_ERROR, "Unable to read block from disk!");
 
-            // Extract the address from the transaction
-            for (unsigned int j = 0; j < tx.vout.size(); j++)
+            for (unsigned int i = 1; i < block.vtx.size(); i++)
             {
                 if (fComplete)
                     break;
 
-                const CTxOut& txout = tx.vout[j];
-                CTxDestination txaddress;
+                // Load Transaction
+                CTransaction tx;
+                CTxDB txdb("r");
+                CTxIndex txindex;
+                uint256 hash;
 
-                // Pass failures here thou we shouldn't have any failures
-                if (!ExtractDestination(txout.scriptPubKey, txaddress))
+                hash = block.vtx[i].GetHash();
+
+                // In case a fail here we can just continue thou it shouldn't happen
+                if (!ReadTxFromDisk(tx, txdb, COutPoint(hash, 0), txindex))
                     continue;
 
-                // If we found a match to multisig address do our work
-                if (CBitcoinAddress(txaddress) == Address)
+                // Extract the address from the transaction
+                for (unsigned int j = 0; j < tx.vout.size(); j++)
```

Comment:
```suggestion
                for (const auto& txout : tx.vout)
```

Reply:
normally I'd agree to that however:
COutPoint requires the tx hash and the vout number
COutPoint does not take a vout directly nor does the vout contain what vout number it is.